### PR TITLE
Base64 various improvements

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6207 ?

Describe changes:
- base64 various improvements

On top of https://github.com/OISF/suricata/pull/9219

### CODE REVIEW

File: src/util-base64.c [hunk 1]
```diff
@@ -83,14 +83,21 @@ bool IsBase64Alphabet(uint8_t encoded_byte)
  *
  * \param ascii the 3-byte ascii output block
  * \param b64 the 4-byte base64 input block
+ * \param nb number of decoded bytes
  *
  * \return none
  */
-static inline void DecodeBase64Block(uint8_t ascii[ASCII_BLOCK], uint8_t b64[B64_BLOCK])
+static inline void DecodeBase64Block(uint8_t ascii[ASCII_BLOCK], uint8_t b64[B64_BLOCK], size_t nb)
```
Comment (catenacyber) [inline src/util-base64.c:95]: Other solution is to ensure `DecodeBase64` is called with a big enough buffer...

File: src/util-base64.c [hunk 2]
```diff
@@ -183,13 +183,14 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
...
 
-    if (bbidx > 0 && bbidx < 4 && ((!valid && mode == BASE64_MODE_RFC4648))) {
+    if (bbidx > 0 && ((!valid && mode == BASE64_MODE_RFC4648))) {
         /* Decoded bytes for 1 or 2 base64 encoded bytes is 1 */
         padding += bbidx > 1 ? B64_BLOCK - bbidx : 2;
         uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
         if (dest_size < *decoded_bytes + numDecoded_blk) {
             SCLogDebug("Destination buffer full");
-            return BASE64_ECODE_BUF;
+            ecode = BASE64_ECODE_BUF;
```
Comment (catenacyber) [inline src/util-base64.c]: That is a question : do we want to decode the maximum ?
But then, we consume bytes that the caller may want to retry...

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15149

**Comment by catenacyber** [comment]
To be reopened after release

**Pull request closed without merging**